src/github.com/getlantern/peerscanner/cf/cf.go

```diff
@@ -0,0 +1,97 @@
+// package cf provides a utility for interacting with CloudFlare
+package cf
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/getlantern/cloudflare"
+	"github.com/getlantern/golog"
+)
+
+var (
+	log = golog.LoggerFor("cf")
+)
+
+type Util struct {
+	Client *cloudflare.Client
+	domain string
+}
+
+func New(domain string, username string, apiKey string) *Util {
+	client := cloudflare.NewClient(username, apiKey)
+	// Set a longish timeout on the HTTP client just in case
+	client.Http = &http.Client{
+		Timeout: 5 * time.Minute,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				ClientSessionCache: tls.NewLRUClientSessionCache(1000),
+			},
+		},
+	}
+	return &Util{client, domain}
+}
+
+func (util *Util) GetRotationRecords(subdomain string) ([]cloudflare.Record, error) {
+	recs, err := util.GetAllRecords()
+	if err != nil {
+		return nil, fmt.Errorf("Could not get records:", err)
+	}
+
+	rotation := make([]cloudflare.Record, 0)
```

Comment:
We could probably add a capacity to `rotation` too, if that represented a performance gain or a clear benefit (if we expected lots of appends or to call this function frequently). Otherwise I agree we could just skip the note.